bigbluebutton-html5/imports/startup/client/intl.jsx

```diff
@@ -39,34 +38,30 @@ class IntlStartup extends Component {
     const url = `/html5client/locale?locale=${locale}`;
 
     const { baseControls } = this.props;
-    this.setState({ appLocale: locale });
 
     baseControls.updateLoadingState(true);
     fetch(url)
       .then((response) => {
-        if (response.ok) {
-          return response.json();
+        if (!response.ok) {
+          return Promise.reject();
         }
-        this.setState({ appLocale: 'en' });
+
         return response.json();
       })
-      .then((messages) => {
-        if (messages.statusCode === 506) {
-          this.setState({ appLocale: 'en' });
-        }
-        this.setState({ messages: messages.messages }, () => {
+      .then(({ messages, normalizedLocale }) => {
+        this.setState({ messages, locale: normalizedLocale.replace('_', '-') }, () => {
           baseControls.updateLoadingState(false);
         });
       })
-      .catch((reason) => {
-        baseControls.updateErrorState(reason);
+      .catch(() => {
+        this.setState({ locale: 'en' });
```

Comment:
We probably should get this from `APP_CONFIG.defaultSettings.application.locale` to match the endpoint.

Reply:
we should